src/util/fileinfo.h

```diff
@@ -0,0 +1,304 @@
+#pragma once
+
+#include <gtest/gtest_prod.h>
+
+#include <QDateTime>
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QUrl>
+#include <QtDebug>
+
+#include "util/assert.h"
+
+namespace mixxx {
+
+/// Helper functions and extensions for QFileInfo using Mixxx terminology.
+namespace fileinfo {
+
+/// Only intended for internal usage.
+inline const QFileInfo& assertNotRelative(const QFileInfo& fileInfo) {
+    // Detect potential, unintended side-effects by the outer context,
+    // i.e. when accessing the transient working directory to resolve
+    // of relative paths. The path should either be empty or absolute,
+    // i.e. not relative.
+    DEBUG_ASSERT(!fileInfo.isRelative());
```

Comment:
Does this function has any value compared to the inlined version? 
I think this moves  the implementation and the comment out of context.
It feels wrong from the name,  that it passes the QFile  through.

Reply:
I didn't want to repeat this documentation 4x in-place. It is crucial to use *is-not-relative* instead of *is-absolute*. Both conditions are not inverse of each other.

Reply:
This can be a comment on top of the class then.

Reply:
If you mind you can add a isAbsolute() member to streamline it a bit.

Reply:
`isAbsolute() != !isRelative()`